sdk/download.go

```diff
@@ -151,3 +154,96 @@ func DownloadSDK(version string) error {
 	tarURL := TarballURL(version)
 	return DownloadSignedFile(tarFile, tarURL)
 }
+
+func fileSum(file string) ([]byte, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	hash := sha512.New()
+	if _, err := io.Copy(hash, f); err != nil {
+		return nil, err
+	}
+
+	return hash.Sum(nil), nil
+}
+
+func compareFileBytes(fileName1, fileName2 string) (bool, error) {
+	sum1, err := fileSum(fileName1)
```

Comment:
Mostly because it was easier to write and easier to get correct. The better way has kinda tricky error conditions. 

Also, when using the verified updater we are only comparing .sig files so the optimization doesn't mean much for the common case. When using UpdateFile() we have to download the image first anyway which dwarfs the file comparison step.

That said after thinking about it more,  I couldn't help writing the optimized version.